pipelines/fetch-farcaster-data/helpers.py

```
@@ -0,0 +1,138 @@
+                        timestamp = datetime.fromisoformat(timestamp_str.replace('Z', '+00:00'))
+                        if timestamp >= cutoff_date:
+                            current_batch.append(item)
+                        else:
+                            reached_cutoff = True
+                
+                if current_batch:
+                    all_data.extend(current_batch)
+                    logging.info(f"Added {len(current_batch)} items within cutoff period")
+                
+                if reached_cutoff:
+                    logging.info(f"Reached cutoff date, stopping pagination")
```

Ruff

f-string without any placeholders

Remove extraneous `f` prefix

(F541)